docs/source/data/errors.mdx

```diff
@@ -482,3 +482,36 @@ In this case, the error above is reported to Apollo Studio as:
 ```
 The REDACTED doesn't have sufficient privileges.
 ```
+
+## HTTP Status Codes
+
+GraphQL, by design, does not use the same conventions from REST to communicate via HTTP verbs and status codes.
+Schema implementors should put information the clients need in the schema, or as part of the standard response `errors` field.
+If possible, consider using the included [Error Codes](#error-codes) or [Custom Errors](#custom-errors) first before modifying http fields.
+
+If you need to have custom behavior, you can set the `http` fields on the response via the [Apollo Server plugins](../integrations/plugins).
+Be aware that GraphQL client libraries may not treat all response status codes the same, and so it will be up to your team to decide what patterns are used.
+
+As an example, here is how you might set a response header and status code based off the GraphQL errors:
+
+
+```js
+const setHttpPlugin = {
+  async requestDidStart() {
+    return {
+      async willSendResponse({ response }) {
+        response.http.headers.set('Custom-Header', 'hello');
+        if (response?.errors?.[0]?.message === 'teapot') {
+          response.http.status = 418;
```

Comment:
Would it be better to only do this if response.http.status is not already set? (Though it might only be set if harsher errors happened like parse errors that aren't likely to lead to response.errors being a thing.)

Reply:
This is just an example. I checked and the status code will not be set yet for a 200 response. I figured this was clear enough logic that you wouldn't take this literally as code you should use seeing as we are returning [HTTP 418](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/418)

Reply:
OK, hmm. There's a lot of complex stuff going into error handling and status codes and odd codepaths, but I think you're right that *most* of the time it's not going to be set yet.